backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java

```diff
@@ -889,68 +806,55 @@ private static void lockedMigration(
-    setDatabaseProperties(App.getInstance(), databasePropMap);
+    // Change liquibase default table names
+    System.getProperties().put("liquibase.databaseChangeLogTableName", "database_change_log");
+    System.getProperties()
+        .put("liquibase.databaseChangeLogLockTableName", "database_change_log_lock");
 
-    if (databaseName.contains("-")) {
-      if (ModelDBHibernateUtil.rDBDialect.equals(ModelDBConstants.POSTGRES_DB_DIALECT)) {
+    // Lock to RDB for now
+    RdbConfig rdb = config.RdbConfiguration;
+
+    if (rdb.RdbDatabaseName.contains("-")) {
+      if (rdb.RdbDialect.equals(ModelDBConstants.POSTGRES_DB_DIALECT)) {
         throw new InterruptedException("Postgres doesn't allow '-' in the database name");
       } else {
-        ModelDBHibernateUtil.createDBIfNotExists(
-            rDBUrl, "`" + databaseName + "`", configUsername, configPassword);
+        ModelDBHibernateUtil.createDBIfNotExists(rdb);
       }
     }
 
     // Check DB is up or not
-    boolean dbConnectionStatus =
-        ModelDBHibernateUtil.checkDBConnection(
-            rDBDriver, rDBUrl, databaseName, configUsername, configPassword, timeout);
+    boolean dbConnectionStatus = ModelDBHibernateUtil.checkDBConnection(rdb, config.timeout);
     if (!dbConnectionStatus) {
       ModelDBHibernateUtil.checkDBConnectionInLoop(true);
     }
 
-    ModelDBHibernateUtil.releaseLiquibaseLock(
-        rDBDriver, rDBUrl, databaseName, configUsername, configPassword);
+    ModelDBHibernateUtil.releaseLiquibaseLock(config);
 
-    String changeSetToRevertUntilTag = (String) databasePropMap.get("changeSetToRevertUntilTag");
     // Run tables liquibase migration
-    ModelDBHibernateUtil.createTablesLiquibaseMigration(
-        rDBDriver, rDBUrl, databaseName, configUsername, configPassword, changeSetToRevertUntilTag);
-
-    LOGGER.info("Liquibase validation stop");
-
-    boolean runLiquibaseSeparate =
-        Boolean.parseBoolean(System.getenv(ModelDBConstants.RUN_LIQUIBASE_SEPARATE));
-    if (runLiquibaseSeparate) {
-      return true;
-    }
-    return false;
+    ModelDBHibernateUtil.createTablesLiquibaseMigration(config, config.changeSetToRevertUntilTag);
   }
 
-  public static void createDBIfNotExists(
-      String rDBUrl, String databaseName, String configUsername, String configPassword)
-      throws SQLException {
+  public static void createDBIfNotExists(RdbConfig rdb) throws SQLException {
 
-    Connection connection = DriverManager.getConnection(rDBUrl, configUsername, configPassword);
+    Connection connection =
+        DriverManager.getConnection(rdb.RdbUrl, rdb.RdbUsername, rdb.RdbPassword);
     ResultSet resultSet = connection.getMetaData().getCatalogs();
 
-    boolean dbExists = false;
     while (resultSet.next()) {
       String databaseNameRes = resultSet.getString(1);
-      if (databaseName.equals("`" + databaseNameRes + "`")) {
-        dbExists = true;
-        break;
+      if (rdb.RdbDatabaseName.equals(databaseNameRes)) {
```

Comment:
```suggestion
      if (rdb.RdbDatabaseName.equals("`" + databaseNameRes + "`")) {
```
to support database name with `-`

Reply:
I believe ` is only necessary when performing a query, right? This is comparing a string provided by the user with a string returned by the db. Neither will have the backtick, afaik.

Reply:
i am not sure about it but if you have tested lets take it in now. i will test it later